api/controllers/inspection.js

```diff
@@ -60,16 +61,33 @@ exports.protectedPostInspection = function (args, res, next) {
   inspection.delete = ['sysadmin', 'inspector'];
   inspection._createdBy = args.swagger.params.auth_payload.preferred_username;
   inspection._createdDate = Date.now();
-  inspection.save()
-    .then(function (theInspection) {
-      Utils.recordAction('Post', 'Inspection', args.swagger.params.auth_payload.preferred_username, theInspection._id);
-      theInspection.status = 'Uploading';
-      return Actions.sendResponse(res, 200, theInspection);
-    })
-    .catch(function (err) {
-      console.log("Error in API:", err);
-      return Actions.sendResponse(res, 400, err);
-    });
+
+  // This is an overwrite in case of error from mobile app,
+  // otherwise this will create a new record.
+  try {
+    // var doc = await Inspection.findOneAndUpdate({ inspectionId: inspection.inspectionId }, inspection, { upsert: true, new: true });
```

Comment:
Can we remove this code if it is not needed?

Reply:
oh woops - missed that. Will push change